lineapy/graph_reader/session_artifacts.py

```diff
@@ -413,7 +433,71 @@ def _slice_session_artifacts(self) -> None:
         )
         self.input_parameters_nodecollection._update_graph(self.graph)
 
+    def _update_nodecollection_dependencies(self):
+        """
+        Update nodecollection dependencies when we replace the cached artifact
+        nodecollection with `lineapy.get`.
+        """
+        last_appearrance_nc: Dict[str, str] = dict()
```

Comment:
comment what this is a map of, my understanding is that this is a map of variable names to the name of the node collection which outputs it.

Reply:
same with dependencies would be nice